src/lib/network_pool/transaction_pool.ml

```diff
@@ -3156,4 +3163,744 @@ let%test_module _ =
+      in
+      let unchanged_major_commands, major_commands_to_merge =
+        split_by_account target_account major_commands
+      in
+      let unchanged_minor_commands, minor_commands_to_merge =
+        split_by_account target_account minor_commands
+      in
+      let%bind major_commands =
+        gen_merge
+          (Array.to_list major_commands_to_merge)
+          (Array.to_list major_sequence)
+          []
+      in
+      let%bind minor_commands =
+        gen_merge
+          (Array.to_list minor_commands_to_merge)
+          (Array.to_list minor_sequence)
+          []
+      in
+      return
+        { prefix_commands
+        ; major_commands =
+            List.append (Array.to_list unchanged_major_commands) major_commands
+            |> List.to_array
+        ; minor_commands =
+            List.append (Array.to_list unchanged_minor_commands) minor_commands
+            |> List.to_array
+        ; minor
+        ; major
+        }
+
+    (** Optional Edge Case : Permission Changes:
+
+        - In major sequence, a transaction modifies an account's permissions:
+            1. It removes the permission to maintain the nonce.
+            2. It removes the permission to send transactions.
+        - In minor sequence, there is a regular transaction involving the same account,
+          but after the permission-modifying transaction in major sequence,
+          the new transaction becomes invalid and must be dropped.
+    *)
+    let gen_updated_branches_for_permission_change
+        { prefix_commands; major_commands; minor_commands; minor; major } =
+      let open Quickcheck.Generator.Let_syntax in
+      let%bind permission_change_cmd =
+        Simple_command.gen_zkapp_blocking_send_and_update_ledger major
+      in
+      let sender_on_major = Simple_command.sender permission_change_cmd in
+      (* We need to increase nonce so transaction has a chance to be placed in the pool.
+         Otherwise it will be dropped as we already have transaction with the same nonce from major sequence
+      *)
+      let sender_index = Simple_ledger.index_of_int sender_on_major.key_idx in
+      let sender_on_minor = Simple_ledger.get minor sender_index in
+      let%bind aux_minor_cmd =
+        Quickcheck_lib.init_gen_array
+          (sender_on_major.nonce - Simple_account.nonce sender_on_minor + 1)
+          ~f:(fun _ ->
+            let sender_on_minor = Simple_ledger.get minor sender_index in
+            let sender_on_minor_idx =
+              Simple_ledger.index_of_int
+                (Simple_account.key_idx sender_on_minor)
+            in
+            Simple_command.gen_single_and_update_ledger minor
+              (sender_on_minor_idx, sender_on_minor) )
+      in
+      return
+        { prefix_commands
+        ; major_commands =
+            Array.append major_commands [| permission_change_cmd |]
+        ; minor_commands = Array.append minor_commands aux_minor_cmd
```

Comment:
Is it possible to mix aux commands (which are expected to be dropped) with the regular commands from `minor_commands`? Or just generate some new random commands and run `gen_merge` on this new sequence and `aux_commands`. We want to ensure that no valid tx from other account is dropped together with aux commands that need to be dropped.